src/providers/CreateQueryViewProvider.ts

```diff
@@ -173,14 +167,28 @@ export class CreateQueryViewProvider
     updateWebviewView(this._context.extensionUri, view, this._activeServerUrl);
   };
 
-  hide = (): void => {
+  hide = async (): Promise<void> => {
     setViewIsVisible(VIEW_ID.createQuery, false);
     showViewContainer(VIEW_CONTAINER_ID.list);
 
-    this._rejectQuerySerial?.(new QueryCreationCancelledError());
+    if (this._querySerialPromiseWithResolvers) {
+      const { promise, reject } = this._querySerialPromiseWithResolvers;
+      this._querySerialPromiseWithResolvers = undefined;
+
+      reject(new QueryCreationCancelledError());
+
+      try {
+        await promise;
+      } catch (err) {}
```

Comment:
Is this necessary? We reject the promise immediately above. You could also just return the promise e.g. `return promise.catch(e => undefined)` I think would do the same (though again - do we actually need it?).